### PR TITLE
quay-enterprise: new doc on configuring QE with a self-signed cert

### PR DESCRIPTION
**Pull request by kbrwn**

### CODE REVIEW

File: quay-enterprise/quay-ssl.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# SSL + Quay Enterprise
+
+**NOTE**: This document assumes you have deployed [Quay Enterprise as a single container.](https://tectonic.com/quay-enterprise/docs/latest/initial-setup.html)
```
Comment (robszumski) [inline quay-enterprise/quay-ssl.md]: Can you take out the "note"? It reads fine without it.

File: quay-enterprise/quay-ssl.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+Quay Enterprise will be configured with a [self-signed certificate.](https://en.wikipedia.org/wiki/Self-signed_certificate) A Certificate Authority is required.
+
+First, create a rootCA:
+
+```
+$ openssl genrsa -out rootCA.key 2048
```
Comment (robszumski) [inline quay-enterprise/quay-ssl.md]: @joshix do we want to start embedding more openssl docs in here?

File: quay-enterprise/quay-ssl.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+$ docker ps
+CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS              PORTS                                                NAMES
+eaf45a4aa12d        quay.io/quay/redis        "/usr/bin/redis-serve"   22 hours ago        Up 22 hours         0.0.0.0:6379->6379/tcp                               dreamy_ramanujan
+cbe7b0fa39d8        quay.io/coreos/registry   "/sbin/my_init"          22 hours ago        Up About an hour    0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 8443/tcp   fervent_ptolemy
+705fe7311940        mysql:5.7                 "/entrypoint.sh mysql"   23 hours ago        Up 22 hours         0.0.0.0:3306->3306/tcp                               mysql
+
+$ docker restart cbe7b0fa39d8
+```
+
+# Confirm
```
Comment (robszumski) [inline quay-enterprise/quay-ssl.md]: confirm what? "Confirm it works"?

Comment (robszumski) [inline quay-enterprise/quay-ssl.md]: or "Test out the secure connection"?

File: quay-enterprise/quay-ssl.md (hunk 1 above)
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: `This document assumes [Quay Enterprise has been deployed as a single container](https://tectonic.com/quay-enterprise/docs/latest/initial-setup.html).`

File: quay-enterprise/quay-ssl.md (hunk 2 above)
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: Yes, for this doc, for now, as discussed in slack.

File: quay-enterprise/quay-ssl.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+```
+$ openssl x509 -req -in ssl.csr -CA rootCA.pem -CAkey rootCA.key -CAcreateserial -out ssl.cert -days 500 -sha256
+Signature ok
+subject=/C=US/ST=California/L=SF/O=Demo Quay/OU=Demo Quay/CN=lan-lab-07.lab.libcore.so/emailAddress=support@quay.io
+Getting CA Private Key
+```
+The next step can be accomplished via the QE superuser panel or from the terminal:
+
+# For setting up via superuser panel:
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: Heading hierarchy, see: https://github.com/coreos/docs/blob/master/STYLE.md#heading-semantics-and-the-sidebar-outline

This is at least an <h2> heading:
`## To configure with the superuser GUI in QE:`

File: quay-enterprise/quay-ssl.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+Save the configuration. QE will automatically validate the SSL certificate.
+
+<img src="img/save-configuration.png" class="img-center" alt="Save and Check config"/>
+
+Restart the container:
+
+<img src="img/restart-container.png" class="img-center" alt="Restart Container"/>
+
+
+# Set up via terminal
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: `## Setting up in the terminal`

File: quay-enterprise/quay-ssl.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+$ docker restart cbe7b0fa39d8
+```
+
+# Test out the secure connection
+
+Confirm the configuration by visiting the URL from a browser `https://lan-lab-07.lab.libcore.so/`
+
+<img src="img/https-browser.png" class="img-center" alt="Browser"/>
+
+"Your Connection is not secure" means the CA is untrusted but confirms that SSL is functioning properly. Google how to trust a CA based
```
Comment (kbrwn) [inline quay-enterprise/quay-ssl.md]: I need help with this line. 

"Google how to trust a CA for the browser/operating system you are running."
"Google how to trust a CA based on the browser/operating system."
Remove the line?

Comment (joshix) [inline quay-enterprise/quay-ssl.md]: "Check Google for how to configure your operating system and browser to trust a certificate signed by your own CA."

Ideally, hand-craft the search string and link to that, or even a great result. But just the sentence is acceptable.

File: quay-enterprise/quay-ssl.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+$ docker ps
+CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS              PORTS                                                NAMES
+eaf45a4aa12d        quay.io/quay/redis        "/usr/bin/redis-serve"   22 hours ago        Up 22 hours         0.0.0.0:6379->6379/tcp                               dreamy_ramanujan
+cbe7b0fa39d8        quay.io/coreos/registry   "/sbin/my_init"          22 hours ago        Up About an hour    0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 8443/tcp   fervent_ptolemy
+705fe7311940        mysql:5.7                 "/entrypoint.sh mysql"   23 hours ago        Up 22 hours         0.0.0.0:3306->3306/tcp                               mysql
+
+$ docker restart cbe7b0fa39d8
+```
+
+# Test out the secure connection
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: Heading hierarchy

File: quay-enterprise/quay-ssl.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+Confirm the configuration by visiting the URL from a browser `https://lan-lab-07.lab.libcore.so/`
+
+<img src="img/https-browser.png" class="img-center" alt="Browser"/>
+
+"Your Connection is not secure" means the CA is untrusted but confirms that SSL is functioning properly. Google how to trust a CA based
+
+# Configuring Docker to Trust a Certificate Authority
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: Heading hierarchy

File: quay-enterprise/quay-ssl.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
+# SSL + Quay Enterprise
+
+This document assumes you have deployed [Quay Enterprise as a single container.](https://tectonic.com/quay-enterprise/docs/latest/initial-setup.html)
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: full stops go outside of the links , e.g., `[Quay Enterprise as a single container](linktarget).

Preferably, write reference-style markdown links, see: https://github.com/coreos/docs/blob/master/STYLE.md#marking-down-the-link

File: quay-enterprise/quay-ssl.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+Quay Enterprise will be configured with a [self-signed certificate.](https://en.wikipedia.org/wiki/Self-signed_certificate) A Certificate Authority is required.
```
Comment (joshix) [inline quay-enterprise/quay-ssl.md]: move the period/full stop outside the link

**Comment by joshix** [comment]
Merging for a few tune ups in a subsequent PR

**Pull request merged**